<?xml version="1.0"?>
<robot name="r2" xmlns:xacro="http://ros.org/wiki/xacro">
	<xacro:property name="PI" value="3.1415926535897931" />
	<xacro:include filename="$(find hector_sensors_description)/urdf/asus_camera.urdf.xacro" />

	<link name="base_link">
		<visual>
			<origin xyz="0 0 -0.13" rpy="0 0 0"/>
			<geometry>
				<mesh filename="package://wild_thumper/meshes/wild_thumper_4wd.stl" scale="1 1 1" />
			</geometry>
		</visual>
	</link>

	<link name="base_footprint">
		<visual>
			<geometry>
				<box size="0.28 0.31 0.0"/>
			</geometry>
			<material name="grey">
				<color rgba="0.5 0.5 0.5 0.5"/>
			</material>
		</visual>
	</link>

	<link name="base_imu_link">
		<visual>
			<geometry>
				<box size="0.041 0.028 0.002"/>
			</geometry>
			<material name="red">
				<color rgba="1 0 0 1"/>
			</material>
		</visual>
	</link>

	<xacro:asus_camera name="camera" parent="base_link">
		<origin xyz="0.107 0.0 0.04" rpy="0 0 0"/>
	</xacro:asus_camera>

	<link name="sonar_forward">
		<visual>
			<geometry>
				<box size="0.016 0.044 0.02"/>
			</geometry>
			<origin xyz="${0.016/2} 0 0" rpy="0 0 0"/>
			<material name="green">
				<color rgba="0 1 0 0.8"/>
			</material>
		</visual>
	</link>

	<link name="sonar_backward">
		<visual>
			<geometry>
				<box size="0.016 0.044 0.02"/>
			</geometry>
			<origin xyz="${0.016/2} 0 0" rpy="0 0 0"/>
			<material name="green">
				<color rgba="0 1 0 0.8"/>
			</material>
		</visual>
	</link>

	<link name="ir_left">
		<visual>
			<geometry>
				<box size="0.015 0.015 0.046"/>
			</geometry>
			<origin xyz="${-0.015/2} 0 0" rpy="0 0 0"/>
			<material name="black">
				<color rgba="0 0 0 1"/>
			</material>
		</visual>
	</link>

	<link name="ir_right">
		<visual>
			<geometry>
				<box size="0.015 0.015 0.046"/>
			</geometry>
			<origin xyz="${-0.015/2} 0 0" rpy="0 0 0"/>
			<material name="black">
				<color rgba="0 0 0 1"/>
			</material>
		</visual>
	</link>

	<joint name="imu_joint" type="fixed">
		<parent link="base_link"/>
		<child link="base_imu_link"/>
		<origin xyz="0.06 -0.03 0.058" rpy="0 0 0"/>
	</joint>

	<joint name="base_link_joint" type="fixed">
		<parent link="base_footprint"/>
		<child link="base_link"/>
		<origin xyz="0.0 0.0 0.13" rpy="0 0 0"/>
	</joint>

	<joint name="sonar_forward_joint" type="fixed">
		<parent link="base_link"/>
		<child link="sonar_forward"/>
		<origin xyz="0.115 0.0 -0.012" rpy="0 0 0"/>
	</joint>

	<joint name="sonar_backward_joint" type="fixed">
		<parent link="base_link"/>
		<child link="sonar_backward"/>
		<origin xyz="-0.115 0.0 -0.012" rpy="0 ${PI} 0"/>
	</joint>

	<joint name="ir_left_joint" type="fixed">
		<parent link="base_link"/>
		<child link="ir_left"/>
		<origin xyz="0.0 ${0.072+0.015} -0.045" rpy="0 0 ${PI/2}"/>
	</joint>

	<joint name="ir_right_joint" type="fixed">
		<parent link="base_link"/>
		<child link="ir_right"/>
		<origin xyz="0.0 ${-0.072-0.015} -0.045" rpy="0 0 ${-PI/2}"/>
	</joint>
</robot>
